src/expr/node_manager.cpp

```diff
@@ -1189,4 +1189,11 @@ Kind NodeManager::getKindForFunction(TNode fun)
   return kind::UNDEFINED_KIND;
 }
 
+Node NodeManager::mkNode(Kind kind, std::initializer_list<TNode> children)
```

Comment:
Are you sure it is safe to take `TNode` here?

Reply:
Yes, it should be ok. The caller ensures that the nodes are alive. I have modified a unit test to pass a temporary node in an init list as a test.